### PR TITLE
refactor(experimental): add getZeroableNullableCodec and getZeroableOptionCodec

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR adds two new`getZeroableNullableCodec` and `getZeroableOptionCodec` functions.

These functions rely on a zero value to represent `None` or `null` values as opposed to using a boolean prefix.

```ts
const codec = getZeroableNullableCodec(getU16Codec());
codec.encode(42); // 0x2a00
codec.encode(null); // 0x0000
codec.decode(new Uint8Array([42, 0])); // 42
codec.encode(new Uint8Array([0, 0])); // null
```

Both functions can also be provided with a custom definition of the zero value using the `zeroValue` option.

```ts
const codec = getZeroableNullableCodec(getU16Codec(), {
    zeroValue: new Uint8Array([255, 255]),
});
codec.encode(42); // 0x2a00
codec.encode(null); // 0xfffff
codec.encode(new Uint8Array([0, 0])); // 0
codec.decode(new Uint8Array([42, 0])); // 42
codec.decode(new Uint8Array([255, 255])); // null
```

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2410?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2410** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2410?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2409** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2409?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2401** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2401?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2400** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2400?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2399** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2399?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2398** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2398?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2397** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2397?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2396** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2396?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2395** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2395?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2394** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2394?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/codecs-data-structures/src/zeroable-nullable.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+function getZeroValue(itemSize: number, zeroValue?: Uint8Array | undefined): Uint8Array {
+    const zeroValueOrDefault = zeroValue ?? new Uint8Array(itemSize).fill(0);
+    if (zeroValueOrDefault.length !== itemSize) {
+        throw new SolanaError(SOLANA_ERROR__CODECS__EXPECTED_ZERO_VALUE_TO_MATCH_ITEM_FIXED_SIZE, {
+            codecDescription: 'zeroableNullable',
+            expectedSize: itemSize,
+            hexZeroValue: getBase16Decoder().decode(zeroValueOrDefault),
+            zeroValue: zeroValueOrDefault,
+        });
```
Comment (steveluscher) [inline packages/codecs-data-structures/src/zeroable-nullable.ts:87]: 👍

**Review by steveluscher — APPROVED** [review]
Added via Giphy

**Comment by lorisleiva** [comment]
### Merge activity

* **Apr 2, 6:48 PM EDT**: @lorisleiva started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2410).

**Pull request merged**